### PR TITLE
test: Replace managed etcd test with generic etcd test

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
Hit #11512

File: test/k8sT/DatapathConfiguration.go [hunk 1]
```diff
@@ -580,14 +580,18 @@ var _ = Describe("K8sDatapathConfig", func() {
...
-	Context("ManagedEtcd", func() {
-		AfterAll(func() {
-			deleteETCDOperator(kubectl)
-		})
-		It("Check connectivity with managed etcd", func() {
+	Context("Etcd", func() {
+		It("Check connectivity", func() {
+			deploymentManager.Deploy(helpers.CiliumNamespace, StatelessEtcd)
+			deploymentManager.WaitUntilReady()
+
```
Comment (aanm) [inline test/k8sT/DatapathConfiguration.go:587]: It seems to me the etcd pod is never removed after this test?

**Review by aanm — CHANGES_REQUESTED** [review]

File: test/k8sT/DatapathConfiguration.go (hunk 1 above)
Comment (tgraf) [reply test/k8sT/DatapathConfiguration.go:587]: It is removed via `deploymentManager.DeleteAll()` in `AfterEach`. This ensures that everything that was deployed via the deployment manager is cleaned up.

**Review by aanm — APPROVED** [review]

**Pull request merged**